docs/user_guide/model_configuration.md

```diff
@@ -321,6 +321,52 @@ configuration](#minimal-model-configuration). You must still provide
 the optional portions of the model configuration by editing the
 config.pbtxt file.
 
+## Custom Model Configuration
+
+Sometimes when multiple devices running Triton instances that share one
+model repository, it is nessecery to have models configured differently
+on each platform in order to achieve the best performance. Triton allows
+users to pick the custom model configuration name by setting `--model-config-name` option.
+
+For example, when running `./tritonserver --model-repository=</path/to/model/repository> --model-config-name=a100`,
+the server will search the custom configuration file `a100.pbtxt` under
+`/path/to/model/repository/<model-name>/configs` directory for each model
+that is loaded. If `a100.pbtxt` exists, it will be used as the configuration
+for this model. Otherwise, the default configuration `/path/to/model/repository/<model-name>/config.pbtxt`
+or [auto-generated model configuration](#auto-generated-model-configuration)
+will be selected based on the settings.
+
+Custom model configuration also works with `Explicit` and `Poll` model
+control modes. Users may delete or add new custom configurations and the
+server will pick the configuration file for each loaded model dynamically.
+
+The repository layout looks like:
+
+```
```

Comment:
I would recommend adding the examples you listed in the TEP instead of doing it this way:

Example 1: --model-config-name=8gpus
.
└── model repository/
    ├── model_a/
    │   ├── configs/
    │   │   ├── 4gpus.pbtxt
    │   │   └── 8gpus.pbtxt
    │   └── config.pbtxt
    ├── model_b/
    │   ├── configs/
    │   │   └── 4gpus.pbtxt
    │   └── config.pbtxt
    └── model_c/
        ├── configs/
        │   └── config.pbtxt
        └── config.pbtxt

Example 2: --model-config-name=config
.
└── model repository/
    ├── model_a/
    │   ├── configs/
    │   │   ├── 4gpu.pbtxt
    │   │   └── 8gpu.pbtxt
    │   └── config.pbtxt
    ├── model_b/
    │   ├── configs/
    │   │   └── 4gpu.pbtxt
    │   └── config.pbtxt
    └── model_c/
        ├── configs/
        │   └── config.pbtxt
        └── config.pbtxt

Example 3: --model-config-name not specified
.
└── model repository/
    ├── model_a/
    │   ├── configs/
    │   │   ├── 4gpu.pbtxt
    │   │   └── 8gpu.pbtxt
    │   └── config.pbtxt
    ├── model_b/
    │   ├── configs/
    │   │   └── 4gpu.pbtxt
    │   └── config.pbtxt
    └── model_c/
        ├── configs/
        │   └── config.pbtxt
        └── config.pbtxt

Reply:
Fixed